freedomofpress/securedrop
# Disable do-release-upgrade OSSEC notification

## Status

Ready for review

## Description of Changes

Fixes #3205

Changes proposed in this pull request:
 * Currently admins are getting weekly alerts encouraging them to upgrade to Xenial, but upgrading to Xenial actually breaks SecureDrop. This addition to the `postinst` scripts of the securedrop OSSEC packages will suppress that notification.

## Testing

These instructions assume that you are testing this in staging VMs.

### Setup

#### Make sure you're getting OSSEC emails

First, since the alert does not appear in vagrant VMs, and it actually only appears once per week, we should address both of those things as setup.

```
git checkout develop
make build-debs
```

Now make sure you have prod like creds and modify your `staging.yml` group vars to enable postfix (i.e. apply [this diff](https://github.com/freedomofpress/securedrop/pull/2803#discussion_r170720578)):

```
vagrant up /staging/
```

Now ensure OSSEC alerts are flowing.

#### Make the `do-release-upgrade` alert happen, and frequently

On each machine, SSH in and in `/etc/update-manager/release-upgrades` replace `Prompt=never` with `Prompt=lts` (this is what is set on hardware). 

Also delete the flag file at `/var/lib/ubuntu-release-upgrader/release-upgrade-available` if it exists.

Finally for easy testing, in `/etc/crontab` run the weekly job every 5 mins: 

```
*/5 * * * * root test -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.weekly)
```

At this point you should be getting the alert about `do-release-upgrade` to your OSSEC email address every 5 minutes. Wait and verify this is the case.

Note: I only ever get the alert from mon on both hardware and prod VMs. However I think it is best to apply this change to both hosts to ensure this alert doesn't pop up in the future for anyone. 

### Test the fix

```
git checkout  resolve-do-release-notification
make build-debs
vagrant provision /staging/
```

- [x] Verify on each server that `/etc/update-manager/release-upgrades` now has `Prompt=never`

At this point, the alert will continue to fire, because the weekly job is running `/usr/lib/ubuntu-release-upgrader/release-upgrade-motd`, and the logic there is to only update the "do we need to do an upgrade" status every 24 hours. This is tracked via the flag file that we deleted earlier, so you need to do one of the following:

1. Wait 24+ hours to wait for the alert to go away 
2. Edit `/usr/lib/ubuntu-release-upgrader/release-upgrade-motd` to expire the flag file more rapidly OR
3. Delete the stamp file at `/var/lib/ubuntu-release-upgrader/release-upgrade-available` so the check (the actual checking for a new release is done in `/usr/lib/ubuntu-release-upgrader/check-new-release`) is performed again (this is what I did)

Once one of these three things is done:

- [x] Verify the alert no longer appears

## Deployment

Will be automatically deployed when SD OSSEC packages update

## Checklist

### If you made changes to the system configuration:


### If you made non-trivial code changes:
